static/src/javascripts/projects/common/modules/identity/upsell/store/consents.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import {
+    getAllConsents,
+    getUserFromApi,
+    setConsent,
+} from 'common/modules/identity/api';
+
+type Consent = {
+    id: string,
+    name: string,
+    description: string,
+    isOptOut: boolean,
+    isChannel: boolean,
+    hasConsented: ?boolean,
+};
+
+const getUserConsents = (): Promise<string[]> =>
+    new Promise(accept => {
+        getUserFromApi(user => {
+            if (user && user.consents) {
+                accept(
+                    user.consents
+                        .filter(consent => consent.consented === true)
+                        .map(consent => consent.id)
+                );
+            } else {
+                accept([]);
+            }
+        });
+    });
+
+const fetchConsents = Promise.all([getUserConsents(), getAllConsents()]);
+
+const get = (): Promise<Consent[]> =>
```

Comment:
Can we use a more descriptive name than `get`? I guess it's scoped by the file name, but I think most call sites will just have to rename this on import (like you did)?

Reply:
yeah, this is a personal preference (generic exports that get renamed) but now that you mention it i think graun uses descriptive names